communication/nats/connection_wrap.go

```diff
@@ -123,14 +114,17 @@ func (c *ConnectionWrap) Close() {
 		c.removeRules()
 	}
 	c.removeRules = nil
+	if c.onClose != nil {
```

Comment:
is this ever nil though? do we need this check?

Reply:
Initialized `onClose` with noop